src/commands.ts

```diff
@@ -365,34 +365,45 @@ export const logs = async (cmd: { json: boolean; open: boolean; setup: boolean;
     if (startDate) {
       filter = `timestamp >= "${startDate.toISOString()}"`;
     }
-    const logs = await logger.entries.list({
-      requestBody: {
-        resourceNames: [`projects/${projectId}`],
-        filter,
-        orderBy: 'timestamp desc',
-      },
-    });
-
-    // We have an API response. Now, check the API response status.
-    spinner.stop(true);
-    // Only print filter if provided.
-    if (filter.length) {
-      console.log(filter);
+    // validate projectId
+    if (!projectId) {
+      return logError(null, ERROR.NO_GCLOUD_PROJECT);
     }
-    if (logs.status !== 200) {
-      switch (logs.status) {
-        case 401:
-          logError(null, oauthSettings.isLocalCreds ? ERROR.UNAUTHENTICATED_LOCAL : ERROR.UNAUTHENTICATED);
-        case 403:
-          logError(
-            null,
-            oauthSettings.isLocalCreds ? ERROR.PERMISSION_DENIED_LOCAL : ERROR.PERMISSION_DENIED,
-          );
-        default:
-          logError(null, `(${logs.status}) Error: ${logs.statusText}`);
+    if (!projectId.match(/^[a-z][a-z0-9\-]{5,29}$/)) {
+      return logError(null, ERROR.PROJECT_ID_INCORRECT(projectId));
+    }
+    try {
+      const logs = await logger.entries.list({
+        requestBody: {
+          resourceNames: [`projects/${projectId}`],
+          filter,
+          orderBy: 'timestamp desc',
+        },
+      });
+      // We have an API response. Now, check the API response status.
+      spinner.stop(true);
+      // Only print filter if provided.
+      if (filter.length) {
+        console.log(filter);
       }
-    } else {
-      printLogs(logs.data.entries);
+      if (logs.status !== 200) {
```

Comment:
It could be nice to have this switch statement in a separate method, like

```
parseResponse = (response) => {
    if (response.status != 200) {
   ...
}
```